includes/class-order.php

```diff
@@ -200,43 +264,272 @@ public static function order_status( $order_id ) {
+		// Check order status.
+		if ( ! in_array( $order->get_status(), array( 'partial-payment', 'overpayment' ), true ) ) {
+			return;
+		}
+
+		// Set totals row label.
+		$label = 'partial-payment' === $order->get_status() ? __( 'Partial Payment', 'zota-woocommerce' ) : __( 'Overpayment', 'zota-woocommerce' );
+
+		?>
+		<table class="wc-order-totals" style="border-top: 1px solid #999; margin-top:12px; padding-top:12px">
+			<tr>
+				<td class="label label-highlight"><?php echo esc_html( $label ); ?>: <br /></td>
+				<td width="1%"></td>
+				<td class="total">
+					<?php echo wc_price( self::get_total_paid( $order ), array( 'currency' => $order->get_currency() ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+				</td>
+			</tr>
+			<tr>
+				<td>
+					<span class="description">
+					<?php
+					if ( ! empty( $order->get_date_paid() ) ) {
+						if ( $order->get_payment_method_title() ) {
+							/* translators: 1: payment date. 2: payment method */
+							echo esc_html( sprintf( __( '%1$s via %2$s', 'zota-woocommerce' ), $order->get_date_paid()->date_i18n( get_option( 'date_format' ) ), $order->get_payment_method_title() ) );
+						} else {
+							echo esc_html( $order->get_date_paid()->date_i18n( get_option( 'date_format' ) ) );
+						}
+					}
+					?>
+					</span>
+				</td>
+				<td colspan="2"></td>
+			</tr>
+		</table>
+
+		<div class="clear"></div>
+		<?php
+	}
+
+
+	/**
+	 * Handle callback.
+	 *
+	 * @param  int                  $order_id Order ID.
+	 * @param  \Zotapay\ApiCallback $callback Callback object.
+	 * @return bool
+	 */
+	public static function handle_callback( $order_id, $callback ) {
+		// Check callback.
+		if ( empty( $callback ) ) {
+			$error = sprintf(
+				// translators: %s WC Order ID.
+				esc_html__( 'Order callback empty for WC Order #%s.', 'zota-woocommerce' ),
+				(int) $order_id
+			);
+			Zotapay::getLogger()->error( $error );
+			return false;
+		}
+
+		$response['status']                 = $callback->getStatus();
+		$response['processorTransactionID'] = $callback->getProcessorTransactionID();
+		$response['errorMessage']           = $callback->getErrorMessage();
+
+		if ( self::amount_changed( $callback ) ) {
+			$extra_data = $callback->getExtraData();
+
+			$response['amountChanged']  = true;
+			$response['amount']         = $callback->getAmount();
+			$response['originalAmount'] = $extra_data['originalAmount'];
+		}
+
+		return self::update_status( $order_id, $response );
+	}
+
+
+	/**
+	 * Handle amount changed
+	 *
+	 * @param  WC_order $order           WC Order.
+	 * @param  array    $response        Response data.
+	 * @return bool
+	 */
+	public static function handle_amount_changed( $order, $response ) {
+		// Convert values to floats.
+		$amount          = \floatval( $response['amount'] );
+		$original_amount = \floatval( $response['originalAmount'] );
+
+		// Add meta.
+		$order->add_meta_data( '_zotapay_amount_changed', sanitize_text_field( $amount ) );
```

Comment:
Suggestion: If `$amount` is already filtered through `floatval()` I think there's no need for `sanitize_text_field` here.